pypulseq/Sequence/sequence.py

```diff
@@ -1460,12 +1461,12 @@ def waveforms_and_times(
                     out_len[-1] += len(rf.t)
 
                     if np.abs(rf.signal[0]) > 0:
-                        pre = np.array([[curr_dur + rf.delay + rf.t[0] - eps], [0]])
+                        pre = np.array([[rf_piece[0, 0] - 50*eps], [0]], dtype=np.complex128)

```

Comment:
1. `rf_piece` is a numpy array, so we can't unpack it with * operator.
2. 50 is an arbitrary number that worked for my case, longer sequences fail even with 50. I tried what you suggest, but eventually, this causes the added time point to have a larger step than raster time, so we have negative time jumps.

Maybe, I should just add, 0.1*rf raster time? My understanding is, we want the time difference to be as small as possible, because, let's say, we integrate the RF or try to do Bloch simulation, adding this point will change the result. Maybe I should ask Dr. Maxim why he added this in the first place.

3. For that line, I don't have to specify, as rf_piece is already complex128. I can remove that.

Reply:
Discussion with Maxim: [https://github.com/pulseq/pulseq/issues/25](https://github.com/pulseq/pulseq/issues/25)

As suspected, proper handling of this is a bit involved. I will use 0.1*rf_raster_time for now, and we can see if it creates a problem later on.